src/middlewared/middlewared/plugins/boot.py

```diff
@@ -257,3 +265,17 @@ def get_swap_size(self, disk):
         Get Automatic Scrub Interval value in days.
         """
         return (await self.middleware.call('system.advanced.config'))['boot_scrub']
+
+
+async def setup(middleware):
+    global BOOT_POOL_NAME
+
+    pools = (
+        await run('zpool', 'list', '-H', '-o', 'name', encoding='utf8')
+    ).stdout.strip().split()
+    for i in BOOT_POOL_NAME_VALID:
+        if i in pools:
+            BOOT_POOL_NAME = i
+            break
+    else:
+        middleware.logger.error('Failed to detect boot pool name.')
```

Comment:
Maybe a middleware shutdown should be performed in this case?

Reply:
I dont think so. Functionality will be limited and things will fail but it will mostly work.